mscore/prefsdialog.cpp

```diff
@@ -250,7 +247,248 @@ PreferenceDialog::PreferenceDialog(QWidget* parent)
+                  new BoolPreferenceItem(PREF_UI_APP_STARTUP_SHOWSPLASHSCREEN, showSplashScreen),
+                  new BoolPreferenceItem(PREF_UI_APP_STARTUP_SHOWSTARTCENTER, showStartcenter),
+                  new BoolPreferenceItem(PREF_UI_APP_STARTUP_SHOWTOURS, showTours),
+                  new BoolPreferenceItem(PREF_APP_TELEMETRY_ALLOWED, collectTelemetry),
+                  new BoolPreferenceItem(PREF_IO_JACK_TIMEBASEMASTER, becomeTimebaseMaster),
+                  new BoolPreferenceItem(PREF_IO_JACK_REMEMBERLASTCONNECTIONS, rememberLastMidiConnections),
+                  new BoolPreferenceItem(PREF_SCORE_NOTE_WARNPITCHRANGE, warnPitchRange),
+                  new StringPreferenceItem(PREF_IMPORT_OVERTURE_CHARSET, importCharsetListOve, nullptr, [&](){ updateCharsetListOve(); }),
+                  new StringPreferenceItem(PREF_IMPORT_GUITARPRO_CHARSET, importCharsetListGP, nullptr, [&](){ updateCharsetListGP(); }),
+                  new DoublePreferenceItem(PREF_SCORE_MAGNIFICATION, scale,
+                                          [&](){ preferences.setPreference(PREF_SCORE_MAGNIFICATION, scale->value()/100.0); },      // apply function
+                                          [&](){ scale->setValue(preferences.getDouble(PREF_SCORE_MAGNIFICATION) * 100.0); }        // update function
+                                                ),
+            #ifdef USE_PORTMIDI
+                  new StringPreferenceItem(PREF_IO_PORTMIDI_INPUTDEVICE, portMidiInput, nullptr, nullptr),
+                  new StringPreferenceItem(PREF_IO_PORTMIDI_OUTPUTDEVICE, portMidiOutput, nullptr, nullptr),
+                  new IntPreferenceItem(PREF_IO_PORTMIDI_OUTPUTLATENCYMILLISECONDS, portMidiOutputLatencyMilliseconds),
+            #endif
+                  new IntPreferenceItem(PREF_EXPORT_AUDIO_SAMPLERATE, exportAudioSampleRate),
+                  new IntPreferenceItem(PREF_EXPORT_MP3_BITRATE, exportMp3BitRate),
+                  new StringPreferenceItem(PREF_SCORE_STYLE_DEFAULTSTYLEFILE, defaultStyle,
+                                          [&]() { // apply function
+                                                preferences.setPreference(PREF_SCORE_STYLE_DEFAULTSTYLEFILE, defaultStyle->text());
+                                                MScore::readDefaultStyle(preferences.getString(PREF_SCORE_STYLE_DEFAULTSTYLEFILE));
+                                                }),
+                  new StringPreferenceItem(PREF_SCORE_STYLE_PARTSTYLEFILE, partStyle,
+                                          [&]() { // apply function
+                                                preferences.setPreference(PREF_SCORE_STYLE_PARTSTYLEFILE, partStyle->text());
+                                                MScore::defaultStyleForPartsHasChanged();
+                                                }),
+                  new StringPreferenceItem(PREF_IMPORT_STYLE_STYLEFILE, importStyleFile,  // is the use of [&] dangerous in this case?
```

Comment:
I think you should try to capture only what is needed whenever you can. In such a big function, capturing everything by reference can mean capturing a lot. In this case it doesn't matter because all it is capturing is `this` and `normalWidgets`, but simply `this` should be enough.